code/modules/clothing/head/_head.dm

```diff
@@ -50,8 +50,11 @@
 		return
 	if(iscyborg(hit_atom))
 		var/mob/living/silicon/robot/R = hit_atom
-		///hats in the borg's blacklist bounce off
-		if(is_type_in_typecache(src, GLOB.blacklisted_borg_hats))
+		var/obj/item/worn_hat = R.hat
+		if(worn_hat && HAS_TRAIT(worn_hat, TRAIT_NODROP))
+			R.visible_message("<span class='warning'>[src] bounces off [worn_hat], without an effect!</span>", "<span class='warning'>[src] bounces off your mighty [worn_hat.name], falling to the floor in defeat.</span>")
+			return
+		if(is_type_in_typecache(src, GLOB.blacklisted_borg_hats))//hats in the borg's blacklist bounce off
```

Comment:
This is intended to be a borg only check, right?

Reply:
Yes, it is intended to be a borg only check. Carbons have their own similar check on line 35.

Reply:
`head` vs `hat`, who would win?